plugins/modules/podman_generate_systemd.py

```diff
@@ -0,0 +1,525 @@
+    if module.params.get('use_names'):
+        command_options.append('--name')
+
+    #  Container-prefix option
+    container_prefix = module.params.get('container_prefix')
+    if container_prefix:
+        command_options.append(f'--container-prefix={container_prefix}')
+
+    #  Pod-prefix option
+    pod_prefix = module.params.get('pod_prefix')
+    if pod_prefix:
+        command_options.append(f'--pod-prefix={pod_prefix}')
+
+    #  Separator option
+    separator = module.params.get('separator')
+    if separator:
+        command_options.append(f'--separator={separator}')
+
+    #  No-header option
+    if module.params.get('no_header'):
+        command_options.append('--no-header')
+
+    #  After option (only for Podman 4.0.0 and above)
+    after = module.params.get('after')
+    if after:
+        for item in after:
+            command_options.append(f'--after={item}')
+
+    #  Wants option (only for Podman 4.0.0 and above)
+    wants = module.params.get('wants')
+    if wants:
+        for item in wants:
+            command_options.append(f'--wants={item}')
+
+    #  Requires option (only for Podman 4.0.0 and above)
+    requires = module.params.get('requires')
+    if requires:
+        for item in requires:
+            command_options.append(f'--requires={item}')
+
+    # Environment variables (only for Podman 4.3.0 and above)
+    environment_variables = module.params.get('env')
+    if environment_variables:
+        for env_var_name, env_var_value in environment_variables.items():
+            command_options.append(
+                f"-e='{env_var_name}={env_var_value}'",
+            )
+
+    #  Full command, with option include
+    command_options.extend(['--format', 'json'])
+    command = [
+        module.params.get('executable'), 'generate', 'systemd',
+        *command_options,
+        module.params['name'],
+    ]
+    command_str = ' '.join(command)
+
+    # Run the podman command to generated systemd .service unit(s) content
+    return_code, stdout, stderr = module.run_command(command)
+
+    # In case of error in running the command
+    if return_code != 0:
+        # Print informations about the error and return and empty dictionary
+        module.fail_json(
+            'Error generating systemd .service unit(s).'
+            f' Command executed: {command_str}'
+            f' Command returned with code: {return_code}.'
+            f' Error message: {stderr}.'
+        )
+
+    # In case of command execution success, its stdout is a json
+    # dictionary. This dictionary is all the generated systemd units.
+    # Each key value pair is one systemd unit. The key is the unit name
+    # and the value is the unit content.
+
+    # Load the returned json dictionary as a python dictionary
+    systemd_units = json.loads(stdout)
+
+    # Write the systemd .service unit(s) content to file(s), if
+    # requested and not in check mode
+    if module.params.get('dest') and not module.check_mode:
```

Comment:
Is `check_mode` processed somewhere?

Reply:
`check_mode` is just "run the module but don't do any modification on the target host" ? Or did I miss something ?

Reply:
In general it should report "changed" is something **would be** changed, but w/o actually doing anything. And to report "ok" when nothing should be changed.

Reply:
If you want to change it you can do it in the next patch.

Reply:
Done.